conjure-go-client/httpclient/client_path_test.go

```diff
@@ -1,85 +0,0 @@
-// Copyright (c) 2019 Palantir Technologies. All rights reserved.
-//
-// Licensed under the Apache License, Version 2.0 (the "License");
-// you may not use this file except in compliance with the License.
-// You may obtain a copy of the License at
-//
-//     http://www.apache.org/licenses/LICENSE-2.0
-//
-// Unless required by applicable law or agreed to in writing, software
-// distributed under the License is distributed on an "AS IS" BASIS,
-// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-// See the License for the specific language governing permissions and
-// limitations under the License.
-
-package httpclient
-
-import (
-	"testing"
-
-	"github.com/stretchr/testify/assert"
-)
-
-func TestJoinURIandPath(t *testing.T) {
-	for _, test := range []struct {
-		baseURI  string
-		reqPath  string
-		expected string
-	}{
-		{
-			"https://localhost",
-			"/api",
-			"https://localhost/api",
-		},
-		{
-			"https://localhost:443",
-			"/api",
-			"https://localhost:443/api",
-		},
-		{
-			"https://localhost:443",
-			"api",
-			"https://localhost:443/api",
-		},
-		{
-			"https://localhost:443/",
-			"api",
-			"https://localhost:443/api",
-		},
-		{
-			"https://localhost:443/foo/",
-			"/api",
-			"https://localhost:443/foo/api",
-		},
-		{
-			"https://localhost:443/foo//////",
-			"////api/",
-			"https://localhost:443/foo/api/",
-		},
-		{
-			"https://localhost:443/foo/",
-			"/api",
-			"https://localhost:443/foo/api",
-		},
-		{
-			"https://localhost",
-			"",
-			"https://localhost",
-		},
-		{
-			"https://localhost/api",
-			"",
-			"https://localhost/api",
-		},
-		{
-			"https://localhost",
-			"/api/+ti%2FQojjmKJxpxmY%2FA=",
-			"https://localhost/api/+ti%2FQojjmKJxpxmY%2FA=",
-		},
-	} {
-		t.Run("", func(t *testing.T) {
-			actual := joinURIAndPath(test.baseURI, test.reqPath)
-			assert.Equal(t, test.expected, actual)
```

Comment:
For posterity/future reference, will note that these test passed with this logic updated to new logic:

```
			baseURL, err := url.Parse(test.baseURI)
			require.NoError(t, err)
			actual := baseURL.JoinPath(test.reqPath).String()
			assert.Equal(t, test.expected, actual)
```

Having verified this once, I don't believe we need to keep these tests any more.